src/Npgsql/Internal/PgSerializerOptions.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Npgsql.Internal.Postgres;
+using Npgsql.NameTranslation;
+using Npgsql.PostgresTypes;
+
+namespace Npgsql.Internal;
+
+public class PgSerializerOptions
```

Comment:
PgConverterContext? We avoid the word Serializer everywhere else..

Reply:
@vonzshik want to weigh in on the name?

I originally had a TODO here stating I would be ok with `PgConverterOptions` too. (Though the functionality of this type really matches a serializer better than the narrower definition of just being about PgConverters)

Reply:
`PgSerializerOptions` is essentially mappings + some info about how to handle specific types + some other settings. While we don't pass it to a serializer (which would have made it much easier), I can't really think of a better name.